app/get_proc_arg.go

```diff
@@ -216,10 +216,40 @@ func getPodLogAsString(kubeClient *clientset.Clientset, namespace, name string)
 	if err != nil {
 		return "", err
 	}
-	logsRes := strings.Split(string(logs), "\n")
-	if strings.HasPrefix(logsRes[0], "No arg") {
-		logrus.Warnf("%s If longhorn driver doesn't work, you may want to try to set argument manually.", logsRes[0])
-		return logsRes[1], nil
+	strLogs := strings.Split(string(logs), "\n")
+	// proc not found
+	if strings.HasPrefix(strLogs[0], "Proc not found") {
+		logrus.Warn(strLogs[0])
+		return "", nil
+	} else if !strings.HasPrefix(strLogs[0], "Proc found") {
+		// unexpected result
+		return "", fmt.Errorf(string(logs))
+	}
+	logrus.Info(strLogs[0])
+	return strLogs[1], nil
+}
+
+func getArgFromCmdline(cmdline string, argNames ...string) string {
+	cmdList := strings.Split(cmdline, "\000")
+	logrus.Infof("Try to find arg %v in cmdline: %v", argNames, cmdList)
+	var arg string
+	for idx, cmdStr := range cmdList {
+		for _, argName := range argNames {
+			if strings.HasPrefix(cmdStr, argName) {
+				if cmdStr == argName {
+					// the separator between <argName> and <argValue> is " "
+					if idx < len(cmdList)-1 {
+						arg = cmdList[idx+1]
+					}
+				} else if strings.Contains(cmdStr, "=") {
+					// the separator between <argName> and <argValue> is "="
+					if !strings.HasSuffix(cmdStr, "=") {
```

Comment:
normally we would run split then check for the element counts. It can avoid the situation like `asd=asd=x`